File: app_watcher/src/main/java/com/tongchen/twatcher/base/http/ResponseFormat.java
package com.tongchen.twatcher.base.http;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Created by TongChen at 14:56 on 2019/08/12.
 * <p>
 * Description: 用于标志解析数据类型的注解
 */

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ResponseFormat {

    String JSON = "json";

    String STRING = "string";

    String XML = "xml";

    String value() default "";
}

File: app_watcher/src/main/java/com/tongchen/twatcher/base/http/ResponseFormat.kt
package com.tongchen.twatcher.base.http

/**
 * @author TongChen
 * @date 2019/12/17  14:06
 * <p>
 * Desc: 用于标志解析数据类型的注解
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class ResponseFormat(val value: String = "") {
    companion object {

        const val JSON = "json"

        const val STRING = "string"

        const val XML = "xml"
    }
}